src/main/java/org/datadog/jenkins/plugins/datadog/apm/JavascriptConfigurator.java

```diff
@@ -0,0 +1,40 @@
+package org.datadog.jenkins.plugins.datadog.apm;
+
+import hudson.FilePath;
+import hudson.model.Node;
+import hudson.model.TaskListener;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+final class JavascriptConfigurator implements TracerConfigurator {
+
+    private static final int GET_NPM_VERSION_TIMEOUT_MILLIS = 30_000;
+    private static final int INSTALL_TRACER_TIMEOUT_MILLIS = 300_000;
+
+    @Override
+    public Map<String, String> configure(DatadogTracerJobProperty<?> tracerConfig, Node node, FilePath workspacePath, Map<String, String> envs, TaskListener listener) throws Exception {
+        String nodeVersion = workspacePath.act(new ShellCommandCallable(Collections.emptyMap(), GET_NPM_VERSION_TIMEOUT_MILLIS, "npm", "-v"));
+        listener.getLogger().println("[datadog] Configuring DD JS tracer: got npm version " + nodeVersion + " from " + workspacePath + " on " + node);
+
+        FilePath datadogPath = workspacePath.child(".datadog");
+        datadogPath.mkdirs();
+
+        // set location for installing global packages (Jenkins agent user may not have the permissions to write to the default one)
+        Path absoluteDatadogPath = Paths.get(datadogPath.absolutize().getRemote());
+        Map<String, String> environment = Collections.singletonMap("NPM_CONFIG_PREFIX", absoluteDatadogPath.toString());
+
+        // we install dd-trace as a "global" package
+        // (otherwise, doing SCM checkout might rollback the changes to package.json, and any subsequent `npm install` calls will result in removing the package)
+        String installTracerOutput = workspacePath.act(new ShellCommandCallable(environment, INSTALL_TRACER_TIMEOUT_MILLIS, "npm", "install", "-g", "dd-trace"));
```

Comment:
Could we have permissions issues?

Reply:
The "global" package naming is a bit misleading here - what this means is we are not installing it inside the repo (because subsequent SCM checkout would overwrite this) and instead we install it inside the `.datadog` folder which we create inside the job's workspace - where we are allowed to write.

We could have a permissions issue if the OS user that runs the Jenkins agent is not allowed to execute `npm`, but I don't think there's anything we can do in this case other than log an error.